akka-actor-testkit-typed/src/main/scala/akka/actor/testkit/typed/scaladsl/LogCapturing.scala

```diff
@@ -62,6 +62,8 @@ trait LogCapturing extends BeforeAndAfterAll { self: TestSuite =>
         s"<-- [${Console.BLUE}${self.getClass.getName}: ${test.name}${Console.RESET}] End of log messages of test that [$res]")
     }
 
+    capturingAppender.clear();
+
```

Comment:
It was actually intended to not clear after each test, because at least the the Cluster tests that we have are often step-by-step and expected to be run in defined order. Then the previous step can convey important information for a later failing test.

In JUnit it's by default not possible to have the test methods depending on each other. Can be done with some annotations. Therefore the log was cleared between each test method.

However, I can understand that many Scala tests are also written independent of each other and then it's good to clear. Can we make this optional? By default it can clear between each test like this but still make it possible to opt out of it.

Reply:
Huh. Good point! I hadn't thought of that. I could see it making sense to keep the current behavior as the  default, especially since it would avoid breaking any existing uses.

I don't think it is possible to add as a parameter to `LogCapturing` since it is a trait. A few options that I think could work:

1. Expose a `clearCapturedLogs` method. If a Scala test suite wants to clear between each test, it can extend `BeforeAndAfterEach` and call `clearCapturedLogs` manually.
2. To make the above easier, there could be a different trait that could either be used with `LogCapturing` or instead of `LogCapturing` that would have the different behavior. Then the configuration would be which version of `LogCapturing` was used.

Do you have a preference or other ideas for how to make it configurable?

Reply:
we could add a `def` to `LogCapturing ` that can be overridden to define the desired behavior
```
protected def clearLogsAfterEachTest: Boolean = false
```

Reply:
That works! I think it can just be a `val` rather than a function. Added that. Also added a method `clearCapturedLogs` for tests that don't want to override that (because they want some mingling between tests) but want to explicitly clear the captured logs (since the capturing appender is internal this allows other use cases to define other behaviors).